### DISCUSSION TITLE
How are people approximating what Azure DevOps templates do?

### DISCUSSION THREAD

**Discussion by devkeydet** [samsmithnz/AzurePipelinesToGitHubActionsConverter#311] [Q&A]
Azure DevOps has templates:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops

GitHub workflow yaml doesn't.  The closest thing I've seen that approximates the behavior is this:
https://github.com/mithro/actions-includes

Anyone else have any other suggested approaches?  Have you considered doing something similar to what actions-includes does in your converter?

**Comment by samsmithnz** [comment]
Apologies for the very slow reply. You are probably on top of this already, but in case others come here with a similar question, there are now reusable workflows in GitHub actions: https://docs.github.com/en/actions/using-workflows/reusing-workflows.

I still haven't worked out if it's worth refactoring this to support them - they work a little differently from Azure Pipeline templates, but this project still does a decent job converting the templates to an action, that can then be refactored into a pipeline. If there is more interest, I can continue to investigate.
